spec/enumerables_spec.rb

```diff
@@ -0,0 +1,211 @@
+    end
+    let(:expected_testArray) { [] }
+
+    it 'Returns a new array substracting 1 from each array' do
+      testArray.my_each_with_index { |_x, i| expected_testArray[i] = testArray[i] - 1 }
+      expect(expected_testArray).to eql([0, 1, 2, 3, 4])
+    end
+
+    it 'Does block instructions on each element (negative)' do
+      testArray.my_each_with_index { |_x, i| expected_testArray[i] = testArray[i] - 1 }
+      expect(expected_testArray).not_to eql(testArray)
+    end
+  end
+
+  describe '#my_select' do
+    let(:expected_testArray) { testArray.my_select(&:even?) }
+
+    it 'Expected all even numbers from [1,2,3,4,5]' do
+      expect(expected_testArray).to eql([2, 4])
+    end
+
+    it 'Expected all even numbers from [1,2,3,4,5' do
+      expect(expected_testArray).not_to eql([])
+    end
+
+    let(:tempArr) { testArray.my_select }
+
+    it 'Return valid enumerable when no block is passed' do
+      expect(tempArr).not_to be_a(Integer)
+    end
+  end
+
+  describe '#my_all?' do
+    let(:tempArr) { testArray.my_all?(&:even?) }
+
+    it 'Return false if not all elements fill criteria' do
+      expect(tempArr).to eql(false)
+    end
+
+    it 'Returns false if not all elements fill criteria (negative)' do
+      expect(tempArr).not_to eql(true)
+    end
+
+    let(:expected_testArray) { testArray.my_all? { |x| x > 0 } }
+
+    it 'Returns true if all elements fill criteria' do
+      expect(expected_testArray).to eql(true)
+    end
+
+    it 'Returns true if all elements fill criteria (negative)' do
+      expect(expected_testArray).not_to eql(false)
+    end
+
+    let(:expected_testArray) { testArray.my_all? }
+
+    it 'Returns true if no block is given and none of the elements are false or nil' do
+      expect(expected_testArray).to eql(true)
+    end
+
+    it 'Returns true if no block is given and none of the elements are false or nil (negative)' do
+      expect(expected_testArray).not_to eql(false)
+    end
+  end
+
+  describe '#my_any?' do
+    let(:tempArr) { testArray.my_any?(&:even?) }
+
+    it 'Returns true if any of the elements fill criteria' do
+      expect(tempArr).to eql(true)
+    end
+
+    it 'Returns true if any of the elements fill criteria (negative)' do
+      expect(tempArr).not_to eql(false)
+    end
+
+    let(:expected_testArray) { testArray.my_any? { |x| x < 0 } }
+
+    it 'Returns false if none of the elements fill criteria' do
+      expect(expected_testArray).to eql(false)
+    end
+
+    it 'Returns false if none of the elements fill criteria (negative)' do
+      expect(expected_testArray).not_to eql(true)
+    end
+  end
+
+  describe '#my_none?' do
+    let(:tempArr) { testArray.my_none?(&:even?) }
+
+    it 'Returns false if any of the elements fill criteria' do
+      expect(tempArr).to eql(false)
+    end
+
+    it 'Returns false if any of the elements fill criteria (negative)' do
+      expect(tempArr).not_to eql(true)
+    end
+
```

Comment:
This also needs to be fixed both are positive case no negative case, check comments above for details